godot-core/src/registry/plugin.rs

```diff
@@ -57,195 +93,470 @@ impl fmt::Debug for ErasedRegisterRpcsFn {
+    ///
+    /// This is mutually exclusive with [`ITraitImpl::user_create_fn`].
+    pub(crate) generated_create_fn: Option<GodotCreateFn>,
+
+    /// Godot low-level `recreate` function, used when hot-reloading a user class.
+    ///
+    /// This is mutually exclusive with [`ITraitImpl::user_recreate_fn`].
+    pub(crate) generated_recreate_fn: Option<
+        unsafe extern "C" fn(
+            p_class_userdata: *mut std::ffi::c_void,
+            p_object: sys::GDExtensionObjectPtr,
+        ) -> sys::GDExtensionClassInstancePtr,
+    >,
+
+    /// Callback to library-generated function which registers properties in the `struct` definition.
+    pub(crate) register_properties_fn: ErasedRegisterFn,
+
+    /// Function called by Godot when an object of this class is freed.
+    ///
+    /// Always implemented as [`callbacks::free`].
+    pub(crate) free_fn: unsafe extern "C" fn(
+        _class_user_data: *mut std::ffi::c_void,
+        instance: sys::GDExtensionClassInstancePtr,
+    ),
+
+    /// Calls `__before_ready()`, if there is at least one `OnReady` field. Used if there is no `#[godot_api] impl` block
+    /// overriding ready.
+    pub(crate) default_get_virtual_fn: Option<
+        unsafe extern "C" fn(
+            p_userdata: *mut std::os::raw::c_void,
+            p_name: sys::GDExtensionConstStringNamePtr,
+        ) -> sys::GDExtensionClassCallVirtual,
+    >,
+
+    /// Whether `#[class(tool)]` was used.
+    pub(crate) is_tool: bool,
+
+    /// Whether the base class is an `EditorPlugin`.
+    pub(crate) is_editor_plugin: bool,
+
+    /// Whether `#[class(internal)]` was used.
+    pub(crate) is_internal: bool,
+
+    /// Whether the class has a default constructor.
+    pub(crate) is_instantiable: bool,
+    #[cfg(all(since_api = "4.3", feature = "register-docs"))]
+    pub(crate) docs: Option<StructDocs>,
+}
+
+impl Struct {
+    pub fn new<T: GodotClass + cap::ImplementsGodotExports>(
+        #[cfg(all(since_api = "4.3", feature = "register-docs"))] docs: Option<StructDocs>,
+    ) -> Self {
+        Self {
+            base_class_name: T::Base::class_name(),
+            generated_create_fn: None,
+            generated_recreate_fn: None,
+            register_properties_fn: ErasedRegisterFn {
+                raw: callbacks::register_user_properties::<T>,
+            },
+            free_fn: callbacks::free::<T>,
+            default_get_virtual_fn: None,
+            is_tool: false,
+            is_editor_plugin: false,
+            is_internal: false,
+            is_instantiable: false,
+            #[cfg(all(since_api = "4.3", feature = "register-docs"))]
+            docs,
+        }
+    }
+
+    pub fn with_generated<T: GodotClass + cap::GodotDefault>(mut self) -> Self {
+        set(&mut self.generated_create_fn, callbacks::create::<T>);
+        #[cfg(since_api = "4.2")]
+        {
+            set(&mut self.generated_recreate_fn, callbacks::recreate::<T>);
+        }
+        self
+    }
```

Comment:
Does this need the `{ }`, or could it just be
```suggestion
    pub fn with_generated<T: GodotClass + cap::GodotDefault>(mut self) -> Self {
        set(&mut self.generated_create_fn, callbacks::create::<T>);
        
        #[cfg(since_api = "4.2")]
        set(&mut self.generated_recreate_fn, callbacks::recreate::<T>);
        
        self
    }
```

Reply:
oh yeah i guess it does, i think it used to be an assignment, and it doesn't work for assignments.